roledb.go

```diff
@@ -495,6 +495,26 @@ func (x *RoleGroupCache) InsertGroup(group *AuthGroup) (err error) {
 	return
 }
 
+func LoadOrCreateGroup(icentral *Central, groupName string, createIfNotExist bool) (*AuthGroup, error) {
```

Comment:
Done.

On 19 January 2015 at 16:12, Ben Harper <notifications@github.com> wrote:

> In roledb.go
> <https://github.com/IMQS/authaus/pull/2#discussion-diff-23161556>:
>
> > @@ -495,6 +495,26 @@ func (x *RoleGroupCache) InsertGroup(group *AuthGroup) (err error) {
> >  	return
> >  }
> >
> > +func LoadOrCreateGroup(icentral *Central, groupName string, createIfNotExist bool) (*AuthGroup, error) {
>
> This function's signature should be:
> func LoadOrCreateGroup(roleDB *RoleGroupDB, groupName string,
> createIfNotExist bool) (*AuthGroup, error)
>
> This stuff is at a "lower" level that Central, and there shouldn't be any
> knowledge of Central down here.
>
> Also, it would be best placed at line 95, because it is a function that
> operates on any object of the interface type RoleGroupDB.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/IMQS/authaus/pull/2/files#r23161556>.
>